tests/subduction_initiation_compositional_fields.prm

```diff
@@ -0,0 +1,87 @@
+# Parameter file for replicating Matsumoto & Tomoda 1983
```

Comment:
Is this a follow-up PR to #5271? If so, once that is major can you update the PRM to have
`include $ASPECT_SOURCE_DIR/cookbooks/subduction_initiation/subduction_initiation_compositional_fields.prm`, and then only specify the parameters that need to be modified?

Otherwise this looks good, thanks!